client/src/features/movie-detail/container.js

```diff
@@ -0,0 +1,99 @@
+import { connect } from 'react-redux';
+import React, { Component } from 'react';
+import { withRouter } from 'react-router-dom';
+import { bindActionCreators } from 'redux';
+import PropTypes from 'prop-types';
+import * as Action from '../common/actions';
+import { getFilmWithId } from '../common/utils';
+import styles from './component.less';
+import Logo from '../../shared/logo';
+import MovieFooter from '../../shared/footer';
+import MovieList from '../../shared/movie-list';
+import Button from '../../shared/button/index';
+
+class MovieDetailContainer extends Component {
+    static propTypes = {
+      location: PropTypes.objectOf(PropTypes.shape({})).isRequired,
+      history: PropTypes.objectOf(PropTypes.shape({})).isRequired,
+    };
+
+    state = {
+      selectFilm: null,
+    };
+
+    componentDidMount() {
+      this.componentChangeItem();
+    }
+
+    componentDidUpdate(prevProps) {
+      const { location } = this.props;
+      if (location.pathname !== prevProps.location.pathname) {
```

Comment:
check `id` route parameter

Reply:
+ I have many doubts that it is possible to dynamically update and `id` in the url... I think you can completely remove this hook